docs/systems.md

```diff
@@ -0,0 +1,126 @@
+# Systems Design Guide
+
+This document describes the design and implementation of the Systems framework in Goose, which enables AI agents to interact with different systems through a unified tool-based interface.
+
+## Core Concepts
+
+### System
+A System represents any component that can be operated by an AI agent. Systems expose their capabilities through Tools and maintain their own state. The core interface is defined by the `System` trait:
+
+```rust
+#[async_trait]
+pub trait System: Send + Sync {
+    fn name(&self) -> &str;
+    fn description(&self) -> &str;
+    fn instructions(&self) -> &str;
+    fn tools(&self) -> &[Tool];
+    async fn status(&self) -> AnyhowResult<HashMap<String, Value>>;
+    async fn call_tool(&self, tool_name: &str, parameters: HashMap<String, Value>) -> ToolResult<Value>;
+}
+```
+
+### Tools
+Tools are the primary way Systems expose functionality to agents. Each tool has:
+- A name
+- A description
+- A set of parameters
+- An implementation that executes the tool's functionality
+
+Tools are defined using a custom attribute macro:
+
+```rust
+#[tool(
+    name = "echo",
+    description = "Echoes back the input"
```

Comment:
missing `parameters: HashMap<String, ToolParameter>`

Reply:
no you're right would have to add that here too